pxr/usd/usdSkel/validators.cpp

```diff
@@ -0,0 +1,91 @@
+//
+// Copyright 2024 Pixar
+//
+// Licensed under the terms set forth in the LICENSE.txt file available at
+// https://openusd.org/license.
+//
+
+#include "pxr/usd/sdr/shaderProperty.h"
+#include "pxr/usd/usd/prim.h"
+#include "pxr/usd/usd/schemaRegistry.h"
+#include "pxr/usd/usd/validationError.h"
+#include "pxr/usd/usd/validationRegistry.h"
+#include "pxr/usd/usd/validator.h"
+#include "pxr/usd/sdr/registry.h"
+#include "pxr/base/tf/token.h"
+#include "pxr/base/tf/stringUtils.h"
+#include "pxr/usd/usdSkel/validatorTokens.h"
+#include "pxr/usd/usdSkel/bindingAPI.h"
+
+#include <algorithm>
+#include <unordered_map>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+static
+UsdValidationErrorVector
+_SkelBindingApiChecker(const UsdPrim &usdPrim)
+{
+    UsdValidationErrorVector errors;
+
+    if (!usdPrim.HasAPI<UsdSkelBindingAPI>()){
+        UsdSchemaRegistry& usdSchemaRegistry = UsdSchemaRegistry::GetInstance();
+        std::unique_ptr<UsdPrimDefinition> primDef = usdSchemaRegistry.BuildComposedPrimDefinition(TfToken(), {TfToken("SkelBindingAPI")});
+
+        std::vector<TfToken> skelPropertyNames = primDef->GetPropertyNames();
+        std::vector<TfToken> primPropertyNames = usdPrim.GetPropertyNames();
+
+        for (const TfToken &primToken : primPropertyNames){
+            for (const TfToken &skelToken : skelPropertyNames){
+                if (skelToken != primToken)
+                    continue;
```

Comment:
Also maybe make skelPropertyNames in unordered_set of Tokens, then you can only iterate on primProperties and do a find of the primProperty in the set, if any is found in the set, then you can report the validation error below and break the for loop! No need to report multiple errors in this case I think.

I think this way it will be more performant. Let me know what you think?

Reply:
i think this is a great idea, implemented.